src/monitor/utils/monitor.factory.ts

```diff
@@ -0,0 +1,59 @@
+import { Monitor } from '../domain/monitor';
+
+export class MonitorFactory {
+  static createMonitor(prismaMonitor): Monitor {
+    const student = {
+      id: prismaMonitor.student.user.id,
+      name: prismaMonitor.student.user.name,
+      contactEmail: prismaMonitor.student.contact_email,
+      whatsapp: prismaMonitor.student.whatsapp,
+      linkedin: prismaMonitor.student.linkedin,
+    };
+
+    const subject = {
+      id: prismaMonitor.subject.id,
+      code: prismaMonitor.subject.code,
+      name: prismaMonitor.subject.name,
+      course: {
+        id: prismaMonitor.subject.course.id,
+        code: prismaMonitor.subject.course.code,
+        name: prismaMonitor.subject.course.name,
+      },
+    };
+
+    const responsible = {
+      id: prismaMonitor.responsible_professor.user.id,
+      name: prismaMonitor.responsible_professor.user.name,
+      email: prismaMonitor.responsible_professor.user.email,
+    };
+
+    const monitorSettings = {
+      id: prismaMonitor.MonitorSettings.id,
+      preferentialPlace: prismaMonitor.MonitorSettings[0].preferential_place,
+      isActive: prismaMonitor.MonitorSettings[0].is_active,
+    };
+
+    const availability = [];
+    for (const availableTime of prismaMonitor.AvailableTimes) {
+      availability.push({
+        id: availableTime.id,
+        week_day: availableTime.week_day,
+        start: availableTime.start,
+        end: availableTime.end,
+      });
+    }
+
+    const monitor: Monitor = {
+      id: prismaMonitor.id,
+      status: prismaMonitor.status.status,
+      endDate: prismaMonitor.end_date,
+      student,
```

Comment:
Aqui você pode usar a Factory de student:

```suggestion
      student: UserFactory.createStudent(prismaMonitor.student),
```

Reply:
usando essa Factory ele mostra outras informações do estudante q não estão especificadas la no card (como `email`, `enrollment`, etc). é isso msm? por isso criei uma classe dentro do domain de Monitor, mas se esse for o comportamento esperado eu importo as outras funcoes sem problemas ^^